presto-hive/src/main/java/com/facebook/presto/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -98,20 +103,23 @@ public class MetastoreHiveStatisticsProvider
     private static final Logger log = Logger.get(MetastoreHiveStatisticsProvider.class);
 
     private final PartitionsStatisticsProvider statisticsProvider;
+    private final QuickStatsProvider quickStatsProvider;
 
-    public MetastoreHiveStatisticsProvider(SemiTransactionalHiveMetastore metastore)
+    public MetastoreHiveStatisticsProvider(SemiTransactionalHiveMetastore metastore, QuickStatsProvider quickStatsProvider)
     {
         requireNonNull(metastore, "metastore is null");
         this.statisticsProvider = (session, table, hivePartitions) -> getPartitionsStatistics(session, metastore, table, hivePartitions);
+        this.quickStatsProvider = quickStatsProvider;
     }
 
     @VisibleForTesting
     MetastoreHiveStatisticsProvider(PartitionsStatisticsProvider statisticsProvider)
     {
         this.statisticsProvider = requireNonNull(statisticsProvider, "statisticsProvider is null");
+        this.quickStatsProvider = null;
```

Comment:
It would be nice to ensure this is never null.  If this is for testing, can we wire in a dummy quick stats provider?